### DISCUSSION TITLE
🐞 registry.dagger.io 503

### DISCUSSION THREAD

**Discussion by bessey** [dagger/dagger#12479] [Help]
### What is the issue?

I am experimenting with Dagger as a first time user. I have been following the official [Python SDK getting started docs](https://docs.dagger.io/sdk/python/628797/get-started). To my surprise, even this mostly empty command is taking >20s to return, and I think its due to an outage on Daggers end.

Bit frustrating as it massively slows down the pace at which I can iterate on my 10% day. Also gives one pause over the dependability of Dagger overall! I would not have expected dagger.io service outage to be able to impact me, though in hindsight it makes sense given the need to download the engine.

### Log output

```
$ time python ci/test.py
failed to resolve image digest: GET https://registry.dagger.io/token?scope=repository%3Aengine%3Apull&service=ghcr.io: unexpected status code 503 Service Unavailable
falling back to leftover engine
#1 resolve image config for docker.io/library/python:3.11-slim-buster
#1 DONE 2.0s

#2 importing cache manifest from dagger:10686922502337221602
#2 DONE 0.0s

#3 from docker.io/python:3.11-slim-buster
#3 resolve docker.io/library/python:3.11-slim-buster
#3 resolve docker.io/library/python:3.11-slim-buster 0.2s done
#3 DONE 0.2s

#4 
Hello from Dagger and Python 3.11.2


________________________________________________________
Executed in   24.54 secs    fish           external
   usr time    2.37 secs   53.00 micros    2.37 secs
   sys time    0.59 secs  552.00 micros    0.59 secs
```

### SDK version

Python SDK v0.3.3

### OS version

macOS 13.2

**Comment by grouville** [comment]
Hi @bessey,

Thanks for your comment 🙏 

Our registry piggybacks to ghcr, which is currently facing a major outage: https://www.githubstatus.com/
<img width="488" alt="image" src="https://user-images.githubusercontent.com/31691250/222166421-c739b28f-6b33-4085-91f2-013bf92a94b0.png">

We are taking that into consideration and have added in our backlog the implementation of a more resilient solution @gerhard 

What you are experiencing is the fix we implemented a few weeks ago after a similar event.
**What is happening is that you hit the timeout issue before we piggyback to the leftover engine:**
```
ghcr.io: unexpected status code 503 Service Unavailable
falling back to leftover engine
```

A short term fix for you would be to run Dagger locally, as a dev version:
From our main branch, at the root of the repo:
```
./hack/dev
```
Then, export the env variables on the terminal from which you run your python SDK script, as printed at the end of the `./hack/dev` script being run:
```
export _EXPERIMENTAL_DAGGER_RUNNER_HOST=docker-container://dagger-engine.dev
export _EXPERIMENTAL_DAGGER_CLI_BIN=/Documents/dagger/bin/dagger
```

**Comment by bessey** [comment]
Haha yep I just realised this myself, when I moved onto evaluating Earthly and `brew` failed too! Cheers, I shall try your workaround.

**Comment by bessey** [comment]
Works for me, cheers.

**Comment by gerhard** [comment]
Re-opening this since @gmlewis hit the same issue earlier today:

<img width="1093" alt="image" src="https://user-images.githubusercontent.com/3342/225417481-42bdf479-f5a5-4ad0-bf52-a6061bc10d6d.png">

I have confirmed that this was due to today's https://www.githubstatus.com/incidents/ybnn77s3lyf8 . I would like to bump this up to the top of my list:

<img width="1118" alt="image" src="https://user-images.githubusercontent.com/3342/225417946-d677023f-4ca7-479b-8b14-6847ca518369.png">

---

FTR https://github.com/dagger/dagger/pull/4517#issuecomment-1422601539

**Comment by gerhard** [comment]
After confirming the root cause, the first action that I took was to modify the existing HTTP probe to alert me when image manifest starts to fail:

![image](https://user-images.githubusercontent.com/3342/225418668-c4910433-efdb-4269-86fc-c1a1c01f93f2.png)

NOOP, just sharing for visibility.

**Comment by celestialorb** [comment]
I was hitting this issue again earlier today due to another GitHub outage. Any plans to allow Dagger to run optionally without the GitHub dependency?
